article.md

```diff
@@ -0,0 +1,612 @@
+
+### Plan:  Schema
+
+- Users: firstName, lastName
+- Posts: AuthorId, title, text, category: enum("tech", "croissants", "techno")
+- Comments: PostId, AuthorId, text
+- PostUpvotes: PostId, UpvoterId
+
+Users (Author) 1:m Posts  
+Users (Author) 1:m Comments  
+Posts 1:m Comments  
+User (Upvoter) n:m Comments
+
+---
+
+## Let's get started
+
+We're about to embark on a rollercoaster adventure. At first, adding Typescript will seem like nothing but a cumbersome waste of time. Hours lost to just littering your Sequelize models with a slew of verbose type annotations. But, peservere! By the time we reach the end, your models will be so well typed that your api routes will practically be writing themselves. Here's a sneak peak.
+
+![image TODO][todo]
+
+Now you're enticed, let's get started. We will be creating a simple JSON api for a blog post website. Apparently, there's a big gap in the market for a site where tech nerds, croissant enthusiasts and techno heads can come together to post blogs. So, our website will allow blogs in these three different categories. Here's the database schema.
+
+> TODO: Probably, in future, change this to a schema picture like the one Martin made for persp.
+
+| Users           | Posts                                          | Comments         | PostUpvotes       |
+|-----------------|------------------------------------------------|------------------|-------------------|
+| id: number      | id: number                                     | id: number       | PostId: number    |
+| name: string    | AuthorId: number                               | PostId: number   | UpvoterId: number |
+| createdAt: date | title: string                                  | AuthorId: number | createdAt: date   |
+| updatedAt: date | text: string                                   | text: string     | updatedAt: date   |
+|                 | category: enum("tech", "croissants", "techno") | createdAt: date  |                   |
+|                 | createdAt: date                                | updatedAt: date  |                   |
+|                 | updatedAt: date                                |                  |                   |
+
+Let's break down the associations. A `User` has many `Post`'s which they have authored. This is the `AuthorId` column on `Posts`. This is a one-to-many relationship. A `User` also has a many-to-many relationship with `Posts` via `PostUpvotes`: a user can upvote many posts, and a post can be upvoted by many users. Finally, `Comments` have both a `User`, via `AuthorId`, and a `Post`. These are both one-to-many.
+
+## Finally, some code
+
+Let's create a simple ExpressJS server. This article will follow along with [this][repo] repo. We use tags so you can see what the code should look
+like at each stage. Let's clone the initial stage, which is just a very simple Express server written in Typescript.
+
+```shell
+$ git clone git@github.com:ahmerb/ts-sqlize-code.git
+$ cd ts-sqlize-code
```

Comment:
I am all for terseness, but I think calling it `sequelize` might search-optimize better. I haven't seen anyone else abbreviate Sequelize to sqlize 🙂

Reply:
Yep, will change this 😃